packages/manager/modules/pci/src/projects/project/private-registry/private-registry.controller.js

```diff
@@ -0,0 +1,94 @@
+import get from 'lodash/get';
+import { PRIVATE_REGISTRY_STATUS, PRIVATE_REGISTRY_STATUS_MAP, GUIDELINK } from './private-registry.constants';
+
+export default class {
+  /* @ngInject */
+  constructor(
+    $state,
+    $stateParams,
+    CucCloudMessage,
+    privateRegistryService,
+  ) {
+    this.$state = $state;
+    this.$stateParams = $stateParams;
+    this.projectId = $stateParams.projectId;
+    this.CucCloudMessage = CucCloudMessage;
+    this.privateRegistryService = privateRegistryService;
+    this.guideLink = GUIDELINK;
+    this.PRIVATE_REGISTRY_STATUS = PRIVATE_REGISTRY_STATUS;
+    this.PRIVATE_REGISTRY_STATUS_MAP = PRIVATE_REGISTRY_STATUS_MAP;
+    this.registryList = [];
+  }
+
+  $onInit() {
+    this.loadMessages();
+    this.getRegistryList(false);
+  }
+
+  loadMessages() {
+    this.CucCloudMessage.unSubscribe('pci.projects.project.private-registry');
+    this.messageHandler = this.CucCloudMessage.subscribe('pci.projects.project.private-registry', { onMessage: () => this.refreshMessages() });
+  }
+
+  refreshMessages() {
+    this.messages = this.messageHandler.getMessages();
+  }
+
+  getRegistryList(clearCache) {
+    return this.privateRegistryService.getRegistryList(this.projectId, clearCache)
+      .then((res) => {
+        this.registryList = res;
+      })
+      .catch(error => this.CucCloudMessage.error(
+        this.$translate.instant('private_registry_query_error', {
+          message: get(error, 'data.message'),
+        }),
+      ));
+  }
+
+  createRegistry() {
+    return this.$state.go('pci.projects.project.private-registry.create', { fromState: 'list' });
```

Comment:
We could use resolves for this.

Reply:
A component could be reused with others action when we want to `createRegistry` for example. In this case it redirects to `pci.projects.project.private-registry.create` but we could do another action in another use of `privateRegistryComponent` component.